src/main/scala/is/hail/methods/IBD.scala

```diff
@@ -172,32 +167,75 @@ object IBD {
         gts.grouped(chunkSize)
           .zipWithIndex
           .map { case (gtGroup, i) => ((i, variantId / chunkSize), (vid, gtGroup)) }
-      }
+    }
       .aggregateByKey(Array.tabulate(chunkSize * chunkSize)((i) => IBSFFI.missingGTCRep))({ case (x, (vid, gs)) =>
         for (i <- gs.indices) x(vid * chunkSize + i) = gs(i)
```

Comment:
this for loop could be optimized. I'm not sure if this is a significant part of the profile, though